setup/ansible_playbook.yml

```diff
@@ -3,55 +3,41 @@
 #
 ---
 
-- name: install dependancies
+- name: Host install & configure
   become: yes
   become_user: root
   hosts: all
   tasks:
 
-  - name: update the apt cache
-    ansible.builtin.apt:
-      update_cache: yes
-      allow_unauthenticated: yes
-
-  - name: install all available apt upgrades
-    ansible.builtin.apt:
-      upgrade: yes
-      allow_unauthenticated: yes
+  - name: Install ROS2 GPG keys
+    ansible.builtin.apt_key: # TODO: retire depracated tool apt-key
+      url: https://raw.githubusercontent.com/ros/rosdistro/master/ros.key
+ 
+  - name: Add ROS2 Humble apt repo
+    ansible.builtin.apt_repository:
+      repo: deb http://packages.ros.org/ros2/ubuntu jammy main
 
-  - name: apt install depends
+  - name: Install components from apt
```

Comment:
the reason gnupg and curl was installed prior to adding the key is because they are needed to install the key

Reply:
I'm not sure what exactly the `apt-key` utility does, but it doesn't call (lib)curl to make it happen. It executes fine on my VM's so it should work on real hardware. There's a high likelihood that we'll want curl on our systems, but it doesn't need to occur prior to ROS2 repo configuration.

Reply:
ok